drizzlepac/haputils/catalog_utils.py

```diff
@@ -2096,25 +2102,18 @@ def detect_segments(self, imgarr, threshold, ncount, filter_kernel=None, source_
 
         """
         log.info("Detecting sources in total image product.")
-        # Insure that regions with negative flux after background subtraction do not
-        # cause the segmentation to create a scene filled with segments, especially with RickerWavelet kernel,
-        # the imgarr needs to be background-subtracted, then clipped at zero before feeding to
-        # 'photutils.detect_sources'.
-        if filter_kernel.min() < 0:  # Found in normalized RickerWavelet kernels
-            img_bkg_sub = imgarr - threshold
-            img_bkg_sub = np.clip(img_bkg_sub, 0, img_bkg_sub.max())
-            # Now set threshold to 0, since it has already been applied to the input data array
-            thresh0 = np.zeros_like(img_bkg_sub)
-        else:
-            # Use inputs provided with the kernel
-            img_bkg_sub = imgarr
-            thresh0 = threshold
 
-        # Note: SExtractor has "connectivity=8" which is the default for detect_sources().
+        # Subtract the background image from the white light image and convolve
+        #
+        # Important to background subtract to get accurate positions for faint sources or
+        # positions will be shifted toward the geometrical center of the segment rather 
+        # than located at the flux-weighted centroid as desired.
+        self.convolved_img = convolve(imgarr-background_img, filter_kernel, normalize_kernel=False)
+
+        # The threshold should not include the background.
         segm_img = None
```

Comment:
This line (`segm_img = None`) is unnecessary now.

Reply:
Indeed as it will be set by the method when no sources are found.  Removed this line.